spec/optional_logger_spec.rb

```diff
@@ -382,4 +382,169 @@
       end
     end
   end
+
+  describe '#info?' do
+    context 'when logger present' do
+      context 'when current severity level allows for info messages' do
+        it 'returns true' do
+          log_content = StringIO.new
+          logger = ::Logger.new(log_content)
+          logger.level = ::Logger::INFO
+          optional_logger = OptionalLogger::Logger.new(logger)
+
+          expect(optional_logger.info?).to eq(true)
+        end
+      end
+
+      context 'when the current severity does not allow for info messages' do
+        it 'returns false' do
+          log_content = StringIO.new
+          logger = ::Logger.new(log_content)
+          logger.level = ::Logger::UNKNOWN
+          optional_logger = OptionalLogger::Logger.new(logger)
+
+          expect(optional_logger.info?).to eq(false)
+        end
+      end
+    end
+
+    context 'when logger NOT present' do
+      it 'returns false' do
+        optional_logger = OptionalLogger::Logger.new(nil)
+        expect(optional_logger.info?).to eq(false)
+      end
+    end
+  end
+
+  describe '#warn?' do
+    context 'when logger present' do
+      context 'when current severity level allows for warn messages' do
+        it 'returns true' do
+          log_content = StringIO.new
+          logger = ::Logger.new(log_content)
+          logger.level = ::Logger::INFO
```

Comment:
While this is true since it's a lower log level than `::LOGGER::WARN`, it doesn't confirm that `::LOGGER::WARN` is sufficient for `#warn?` to return true.